src/sql/platform/connection/common/connectionManagementService.ts

```diff
@@ -543,10 +549,46 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		if (options.showDashboard) {
 			this.showDashboardForConnectionManagementInfo(connectionManagementInfo.connectionProfile);
 		}
+
+		let connectionProfile = connectionManagementInfo.connectionProfile;
 		this._onConnect.fire(<IConnectionParams>{
 			connectionUri: uri,
-			connectionProfile: connectionManagementInfo.connectionProfile
+			connectionProfile: connectionProfile
 		});
+
+		let iconProvider = this._iconProviders.get(connectionManagementInfo.providerId);
+		if (iconProvider) {
+			let serverInfo: azdata.ServerInfo = this.getServerInfo(connectionProfile.id);
+			let iConnectionProfile: azdata.IConnectionProfile = this.toIConnectionProfile(connectionProfile);
+			iconProvider.getConnectionIconId(iConnectionProfile, serverInfo).then(iconId => {
+				if (iconId) {
+					this._connectionIconIdCache.set(connectionProfile.id, iconId);
+				}
+			});
+		}
+	}
+
+	public getConnectionIconId(connectionId: string): string {
+		return this._connectionIconIdCache.get(connectionId);
+	}
+
+	private toIConnectionProfile(cf: ConnectionProfile): azdata.IConnectionProfile {
```

Comment:
The .toIConnectionProfile method already exists inside the `ConnectionProfile` class. Please use that and delete this method

Reply:
fixed.